lib/src/http.dart

```diff
@@ -325,10 +333,11 @@ Future<T> retryForHttp<T>(String operation, FutureOr<T> Function() fn) async {
   return await retry(
     () async => await _httpPool.withResource(() async => await fn()),
     retryIf: (e) async =>
-        (e is PubHttpException && e.isIntermittent) ||
-        e is TimeoutException ||
-        e is http.ClientException ||
-        isHttpIOException(e),
+        !_pubClient._wasClosed &&
```

Comment:
Let's make a `class ClientClosedException implements ClientException` which indicates that the request failed, because the client was closed while the request was in progress. Such exceptions should not be retried.

Reply:
Yeah that's better.

Done (though I avoided implementing `ClientException`).